### PR TITLE
refactor: do code refactor - part1

### PR DESCRIPTION
**Pull request by JPedroBorges**
Part 1: refactor manager and dependencies

Separate concerns between key manager and extension store.

### CODE REVIEW

File: src/handler/operation/remote.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        for (let i = 0; i < sessionItems.length; i++) {
+            try {
+                let value = window.localStorage.getItem(sessionItems[i].key);
+
+                if (sessionItems[i].subKey) {
+                    value = JSON.parse(value)[sessionItems[i].subKey];
+                }
+
+                sessionItems[i].value = value;
+            } catch (e) { }
```
Comment (Marianarodrigues7) [inline src/handler/operation/remote.js:21]:
```suggestion
            } catch () { }
```

File: src/handler/operation/remote.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                    newValue = {};
+                }
+                newValue[subKey] = value;
+            }
+
+            const stringifiedValue =
+                newValue instanceof Object ? JSON.stringify(newValue) : newValue;
+
+            window.sessionStorage.setItem(key, stringifiedValue);
+        } catch (e) {
```
Comment (Marianarodrigues7) [inline src/handler/operation/remote.js:59]:
```suggestion
        } catch () {
```

File: src/handler/store.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        storeItems = changes.storeItems.newValue;
+        onStoreUpdate(storeItems);
+    });
+
+    async function getItems() {
+        try {
+            const storeResult = await chrome.storage.local.get(["storeItems"]);
+            storeItems = storeResult["storeItems"];
+            return storeItems;
```
Comment (Marianarodrigues7) [inline src/handler/store.js:16]:
```suggestion
            return storeResult["storeItems"];
            
```

File: src/handler/valueReader.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+function valueReader(remote, cookie) {
+  async function getSessionValues(sessionStoreItems) {
+    try {
+      return await remote.executeRequest(
+        [sessionStoreItems],
+        remote.getSessionValue
+      );
+    } catch (e) {
+      console.log("🐶 ~ could not read local values:", e);
```
Comment (Marianarodrigues7) [inline src/handler/valueReader.js:9]:
```suggestion
      console.log("🐶 ~ could not read session values:", e);
```

File: src/manager.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    tab = result;
+
+    remoteExecutor = remote(tab);
+    cookieExecutor = cookie(tab);
+    reader = valueReader(remoteExecutor, cookieExecutor);
+    writer = valueWriter(remoteExecutor, cookieExecutor);
+  });
+
+  async function getPresentationItems(storeItems) {
+    let presentationItems = [];
```
Comment (Marianarodrigues7) [inline src/manager.js:18]:
```suggestion
    const presentationItems = [];
```

File: src/popup.html [hunk 6]
```diff
@@ -0,0 +1,192 @@
+<!-- popup.html -->
```
Comment (Marianarodrigues7) [inline src/popup.html:1]: No need to have this

File: manifest.json [hunk 7]
```diff
@@ -9,7 +9,7 @@
       "128": "images/icon128.png"
     },
     "default_title": "Click Me",
-    "default_popup": "popup.html"
+    "default_popup": "src/popup.html"
   },
   "description": "the auth retriever",
```
[the remark this answers is not in the document]
Comment (JPedroBorges) [reply manifest.json]: was already done :)

File: src/popup.html (hunk 6 above)
Comment (JPedroBorges) [reply src/popup.html:1]: done.

**Review by Marianarodrigues7 — APPROVED** [review]
Great job 💪🏻

**Pull request merged**